### PR TITLE
Created AppleScript Cheat Sheet

### PR DESCRIPTION
**Pull request by J-tt**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [x] New Instant Answer
  - [x] Cheat Sheets: **`New AppleScript Cheat Sheet`**
  - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
  - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
  - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**
###### Description of changes

Created an instant answer cheat sheet for AppleScript, using the [Mac Developer Library](https://developer.apple.com/library/mac/documentation/AppleScript/Conceptual/AppleScriptLangGuide/introduction/ASLR_intro.html#//apple_ref/doc/uid/TP40000983-CH208-SW1)
###### Which issues (if any) does this fix?
#2986

https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage
###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/applescript_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @J-tt

### CODE REVIEW

**Comment by daxtheduck** [comment]
@J-tt Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=AppleScript Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/applescript_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=AppleScript Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/applescript_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=AppleScript Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/applescript_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=AppleScript Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/applescript_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=AppleScript Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/applescript_cheat_sheet_mobile.png)

**Comment by mintsoft** [comment]
I don't have access to OSX can anyone else review this @duckduckgo/duckduckhack-contributors

**Comment by ethanchewy** [comment]
@mintsoft I have OSX. I'll look at it tonight!

**Comment by J-tt** [comment]
Thanks guys!

**Comment by ethanchewy** [comment]
I think there are a lot of issues with this cheatsheet. With all due respect, the only good part I see is the commands section.

Here are some of the issues with this cheatsheet IMO:
- The variable and operator sections are unnecessary.Those parts of AppleScript are quite familiar to programmers. 
- "list" and "record" are types of mutable objects. Perhaps create a "mutable object" section instead
- The "repeat" section is quite messy. Also, it should be under "Control Statements". See: [](https://developer.apple.com/library/mac/documentation/AppleScript/Conceptual/AppleScriptLangGuide/reference/ASLR_control_statements.html)

@moollaza Am I being too harsh? I think this cheatsheet has started in the wrong angle.

@J-tt Consider In addition to what I said above, consider adding sections like "Script Objects", "Apple Keywords", "Class Reference".

**Comment by mintsoft** [comment]
@ethanchewy I'm going to assign you to this for now, just gimme a ping etc if you need a hand/it's ready for merge

File: share/goodie/cheat_sheets/json/applescript.json [hunk 1]
```diff
@@ -0,0 +1,160 @@
+{
+    "id": "applescript_cheat_sheet",
+    "name": "AppleScript Cheat Sheet",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/applescript.json]: Lets change the name to "AppleScript"

File: share/goodie/cheat_sheets/json/applescript.json [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    "description": "A cheat sheet for AppleScript",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/applescript.json]: Lets make a description which will describe the contents of the cheat sheet, else remove it

File: share/goodie/cheat_sheets/json/applescript.json [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            }
+        ],
+        "Control Statements": [
+            {
+                "key": "if boolean then statement",
+                "val": "Executes statement if boolean is true"
+            },
+            {
+                "key": "repeat",
+                "val": "repeat a statement"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/applescript.json]: Capitalize first letter at val field

File: share/goodie/cheat_sheets/json/applescript.json [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                "val": "Launches the specified application without sending it a run command."
+            },
+             {
+                "key": "run",
+                "val": "For an application, launches it. For a script application, launches it and sends it the run command. For a script script object, executes its run handler."
+            },
+             {
+                "key": "set",
+                "val": "Assigns one or more values to one or more script variables or application objects."
+            }
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/applescript.json]: remove the periods at end of val fields

**Comment by pjhampton** [comment]
Closing as this PR looks to have gone stale.

If you are looking for something more substantial to work through that will have a big impact for the users of DuckDuckGo, please check out the [DuckDuckHack Issues Table](https://duckduckhack.com/issues). 🚀  😄

**Pull request closed without merging**